monitor.py

```diff
@@ -265,7 +265,7 @@ def check_disk_space(paths):
 
 
 # Make sure we can connect (might prompt for username/password)
-if subprocess.call(['cl', 'work', 'localhost::']) != 0:
+if subprocess.call(['cl', 'work']) != 0:
```

Comment:
Why didn't `localhost::` work?  `localhost` should be a default alias...

Reply:
This is what I got when `localhost::` specified:
```
process: [2019-08] >> timeout 60s cl work localhost:: (exit code 1, time 0.50s [limit: 15s,60s]; avg 0.50s; max 0.50s)
process: [2019-08] Traceback (most recent call last):
process: [2019-08]   File "/usr/local/bin/cl", line 11, in <module>
process: [2019-08]     load_entry_point('codalab', 'console_scripts', 'cl')()
process: [2019-08]   File "/opt/codalab-worksheets/codalab/bin/cl.py", line 117, in main
process: [2019-08]     cli.do_command(sys.argv[1:])
process: [2019-08]   File "/opt/codalab-worksheets/codalab/lib/bundle_cli.py", line 843, in do_command
process: [2019-08]     structured_result = command_fn()
process: [2019-08]   File "/opt/codalab-worksheets/codalab/lib/bundle_cli.py", line 837, in <lambda>
process: [2019-08]     command_fn = lambda: args.function(self, args)
process: [2019-08]   File "/opt/codalab-worksheets/codalab/lib/bundle_cli.py", line 2779, in do_work_command
process: [2019-08]     client, worksheet_uuid = self.parse_client_worksheet_uuid(args.worksheet_spec)
process: [2019-08]   File "/opt/codalab-worksheets/codalab/lib/bundle_cli.py", line 707, in parse_client_worksheet_uuid
process: [2019-08]     worksheet_uuid = self.resolve_worksheet_uuid(client, base_worksheet_uuid, spec)
process: [2019-08]   File "/opt/codalab-worksheets/codalab/lib/bundle_cli.py", line 635, in resolve_worksheet_uuid
process: [2019-08]     'worksheets', params={'base': base_worksheet_uuid, 'specs': [worksheet_spec]}
process: [2019-08]   File "/opt/codalab-worksheets/codalab/client/json_api_client.py", line 425, in fetch_one
process: [2019-08]     results = self.fetch(resource_type, resource_id=resource_id, params=params)
process: [2019-08]   File "/opt/codalab-worksheets/codalab/client/json_api_client.py", line 50, in wrapper
process: [2019-08]     sys.exc_info()[2],
process: [2019-08]   File "/opt/codalab-worksheets/codalab/client/json_api_client.py", line 16, in wrapper
process: [2019-08]     return f(*args, **kwargs)
process: [2019-08]   File "/opt/codalab-worksheets/codalab/client/json_api_client.py", line 415, in fetch
process: [2019-08]     query_params=self._pack_params(params),
process: [2019-08]   File "/opt/codalab-worksheets/worker/codalabworker/rest_client.py", line 91, in _make_request
process: [2019-08]     with closing(urllib2.urlopen(request)) as response:
process: [2019-08]   File "/usr/lib/python2.7/urllib2.py", line 154, in urlopen
process: [2019-08]     return opener.open(url, data, timeout)
process: [2019-08]   File "/usr/lib/python2.7/urllib2.py", line 429, in open
process: [2019-08]     response = self._open(req, data)
process: [2019-08]   File "/usr/lib/python2.7/urllib2.py", line 447, in _open
process: [2019-08]     '_open', req)
process: [2019-08]   File "/usr/lib/python2.7/urllib2.py", line 407, in _call_chain
process: [2019-08]     result = func(*args)
process: [2019-08]   File "/usr/lib/python2.7/urllib2.py", line 1228, in http_open
process: [2019-08]     return self.do_open(httplib.HTTPConnection, req)
process: [2019-08]   File "/usr/lib/python2.7/urllib2.py", line 1198, in do_open
process: [2019-08]     raise URLError(err)
process: [2019-08] codalab.client.json_api_client.JsonApiException: Unable to fetch worksheets: <urlopen error [Errno 99] Cannot assign requested address>
process: [2019-08]
```

Reply:
With this logic : https://github.com/codalab/codalab-worksheets/blob/master/monitor.py#L268-L269, monitor.py won't run once it throws an error.

Reply:
I was trying to figure out why the error was happening.  I think the problem is that `localhost::` points to http://localhost, but inside docker, you need to use `http://rest-server`.  The `localhost::` was probably there for a reason, but I think this case it's probably fine.

Reply:
The thing is that if you run `cl work` within the docker directly or `docker exec -it codalab_rest-server_1 cl work localhost::`, both of them worked fine. But with the wait-for-it.sh wrapper, it failed. I tested `cl work`, it seems default to localhost...